### PR TITLE
[macOS] Fix generated bundle name for templates

### PR DESCRIPTION
**Pull request by adamscott**
Currently, if you build using the following scons commands, both of these will result in generating `bin/godot_macos_dev.zip`.

```console
scons platform=macos target=template_debug generate_bundle=yes
scons platform=macos target=template_release generate_bundle=yes
```

Unfortunately, it makes it so that if you're building the editor and the templates at the same time, the latter built template will overwrite the former one.

```console
# The template debug artifact will be overwritten.
scons platform=macOS target=editor generate_bundle=yes && scons platform=macos target=template_debug generate_bundle=yes && scons platform=macos target=template_release generate_bundle=yes
```

This PR adds a prefix to the generated template bundle (the zip file).

Capture d’écran, le 2024-10-29 à 13 11 58

_**TL;DR** Instead of generating 2 times the highlighted file, the previous command will generate the 2 files on top instead._

### CODE REVIEW

**Review by bruvzg — COMMENTED** [review]
> Currently, if you build using the following scons commands, both of these will result in generating bin/godot_macos_dev.zip.

I think this is expected, `generate_bundle` pack both release and debug templates to a single (this is how official templates are built), so you should be able to use it only once:

```
scons platform=macos target=template_debug
scons platform=macos target=template_release generate_bundle=yes
```

**Comment by bruvzg** [comment]
> that a build expects the side-effects from another build to generate a bundle.

Also, it will be affected by multiple builds anyway, do to multiple archs and making universal binary.

**Comment by adamscott** [comment]
Closing per @bruvzg comments.
